### PR TITLE
[Readm] Update to 1.0.10

### PR DESCRIPTION
**Pull request by TheNetsky**
- Added id checking to all mangaId parsing.
- Updated chapterId parsing method, site added /all-pages param by default on some titles
- Replaced some split methods with replace methods
- Added id and label checking for tags
- Added failback image for mangaDetails
- Added regex testing
- Fixed parsing error with chapters on some edge cases (Some titles have no chapter numbers at all! Just "Vol" or "Chapter', these will now default to 0 properly)
- Updated test file

### CODE REVIEW

**Review by ConradWeiser — CHANGES_REQUESTED** [review]
Looks very good! I've only caught a few things which may need addressed in the Search Parsing function.
Thanks for your efforts!

File: src/Readm/ReadmParser.ts [hunk 1]
```diff
@@ -63,11 +64,13 @@ export const parseMangaDetails = ($: CheerioStatic, mangaId: string): Manga => {
 export const parseChapters = ($: CheerioStatic, mangaId: string): Chapter[] => {
   const chapters: Chapter[] = [];
 
-  for (const elem of $("div.season_start").toArray()) {
-    const title = $("h6.truncate", elem).first().text().trim() ?? "";
-    const chapterId = $('a', elem).attr('href')?.split(mangaId + "/").pop() ?? '';
-    const chapterNumber = Number(title.match(/(\d+)/)![0] ?? 0);
-    const date = parseDate($("td.episode-date", elem)?.text() ?? "");
+  for (const c of $("div.season_start").toArray()) {
```
[the remark this answers is not in the document]
Comment (TheNetsky) [reply src/Readm/ReadmParser.ts:67]: Appreciate the feedback on this one, I wasn't too sure. In the past I used different variables all over the place. So I thought I'd change it some the same so it looks a bit cleaner. 

I think using some longer context ones is the way to go in the future. It's easy to change variables once everything has been completed. But during development the longer ones make it more readable. 👍

File: src/tests/Readm.test.ts [hunk 2]
```diff
@@ -86,6 +87,16 @@ describe('Readm Tests', function () {
...
+    });
+    it("Testing home page results for new titles", async () => {
+        let results = await wrapper.getViewMoreItems(source, "new_manga", {}, 1)
+
+        expect(results, "No results whatsoever for this section").to.exist
+        expect(results, "No results whatsoever for this section").to.exist
+
+        let data = results![0]
+        expect(data.id, "No ID present").to.exist
+        expect(data.image, "No image present").to.exist
```
Comment (AdrienSeon) [inline src/tests/Readm.test.ts:99]: This test is not needed since there is no View more for this section.

File: src/tests/Readm.test.ts [hunk 3]
```diff
@@ -94,30 +105,29 @@ describe('Readm Tests', function () {
...
         expect(result.subtitleText, "No subtitle text").to.be.not.null;
     });
-
     it("Testing Home-Page aquisition", async () => {
         let homePages = await wrapper.getHomePageSections(source)
         expect(homePages, "No response from server").to.exist
         expect(homePages[0], "No top weekly section available").to.exist
         expect(homePages[1], "No latest updates section available").to.exist
         expect(homePages[2], "No new manga section available").to.exist
     });
```
Comment (AdrienSeon) [inline src/tests/Readm.test.ts:121]: This test doesn't seem up to date with the homepage structure.

File: src/tests/Readm.test.ts [hunk 4]
```diff
@@ -55,8 +55,8 @@ describe('Readm Tests', function () {
         expect(data.mangaId, "Missing MangaID").to.be.not.empty;
         expect(data.pages, "No pages present").to.be.not.empty;
     });
-    it("Testing home page results for hot titles", async () => {
-        let results = await wrapper.getViewMoreItems(source, "hot_manga", {}, 1)
+    it("Testing home page results for hot update titles", async () => {
```
Comment (AdrienSeon) [inline src/tests/Readm.test.ts:58]: This test is not needed since there is no View more for this section.

**Review by AdrienSeon — APPROVED** [review]
Tested in-app, working great!
Some really minor test file cleanups but nothing blocking
Thanks for your work!

**Review by AdrienSeon — APPROVED** [review]

**Review by ConradWeiser — APPROVED** [review]
